packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf

```diff
@@ -0,0 +1,74 @@
+resource "random_string" "api_gateway_service_account_id" {
+  length  = 20
+  lower   = true
+  upper   = false
+  special = false
+  number  = false
+}
+
+resource "google_service_account" "api_gateway_service_account" {
+  account_id   = random_string.api_gateway_service_account_id.result
+  display_name = "${var.name}-service-account"
+}
+
+resource "google_project_iam_member" "api_gateway_monitoring_writer_role" {
+  role    = "roles/monitoring.metricWriter"
+  member  = "serviceAccount:${google_service_account.api_gateway_service_account.email}"
+  project = var.project
+}
+
+resource "google_project_iam_member" "api_gateway_logging_writer_role" {
+  role    = "roles/logging.logWriter"
+  member  = "serviceAccount:${google_service_account.api_gateway_service_account.email}"
+  project = var.project
+}
+
+resource "google_cloudfunctions_function_iam_member" "invoker" {
+  for_each = toset(var.invoke_targets)
+
+  cloud_function = each.key
+  role           = "roles/cloudfunctions.invoker"
+  member         = "serviceAccount:${google_service_account.api_gateway_service_account.email}"
+  project        = var.project
+}
+
+resource "google_api_gateway_api" "api_gateway_api" {
+  provider = google-beta
+
+  api_id       = lower("${var.name}-api")
+  display_name = "${var.name}-api"
+}
+
+resource "google_api_gateway_api_config" "api_gateway_api_config" {
+  provider = google-beta
+
+  api           = google_api_gateway_api.api_gateway_api.api_id
+  # Generating a suffix manually because the one generated with `api_config_id_prefix` is far too long
+  api_config_id = lower("${var.name}-config-${substr(uuid(), 0, 8)}")
```

Comment:
This is used to uniquely identify an API config. During the deployment of a new version, first, a new config is created and then the old one is destroyed, so, for a few moments, there are two of them, that's why we need to distinguish them.
The ID can have a max length of 64 characters. Taking into consideration our naming convention, I don't really want to make the identifying part longer as it may become a problem. But anyway, the first 8 characters from a UUID should be unique enough to avoid collisions between just two resources.